### PR TITLE
RFC: Updates and Firefox

### PR DESCRIPTION
**Pull request by dakra**
This is based on #128.
I reindented all javascript files and updated jQuery so it's
in theory ready for Firefox (and Edge?) addon store.

I made a zip for anyone who wants to try it out,
since it's unsigned you have to set
` xpinstall.signatures.required` to `true` in `about:config`
(which only works in the developer edition or nightly).
[emacs_chrome.zip](https://github.com/stsquad/emacs_chrome/files/1132708/emacs_chrome.zip)

### CODE REVIEW

File: javascript/textareas.js [hunk 1]
```diff
@@ -373,29 +367,29 @@ function localMessageHandler(msg, port) {
...
             callback: handleUpdatedElements,
             queries: [
-                {   // we don't want the source page accidently duplicating our tags
-                    element: "img[class='ewe_edit_button']"
+                {
+                    // we don't want the source page accidently duplicating our tags
+                    element: "img[class='ewe_edit_button']",
                 },
                 {
-                    element: "textarea"
```
Comment (stsquad) [inline javascript/textareas.js]: Why is the quoting on this changed but not the others?

File: javascript/xmlcomms.js [hunk 2]
```diff
@@ -133,12 +133,6 @@ function handleContentMessages(msg, tab_port) {
                         'Unable to contact an edit server, is it running?' +
                             " I'll take you to the options page when you close this",
                 });
-                notification = chrome.notifications.createNotification(
```
Comment (stsquad) [inline javascript/xmlcomms.js]: Hmm I don't see this in master, has one of the commits introduced it?

File: manifest.json [hunk 3]
```diff
@@ -20,7 +20,10 @@
 	{
 		"128" : "icons/emacs.png"
 	},
-	"options_page": "fancy-settings/source/index.html",
```
Comment (stsquad) [inline manifest.json]: This seems un-related to the API changes

**Comment by stsquad** [comment]
Hi,

Thank you for your PR. As you no doubt intended it as an RFC you'll hopefully not be surprised that I won't merge it as is. I'm a little on the fence about global changes to javascript formatting as the commit tends make tracing history a pain. I agree the formatting is a mess though and I'd like to fix it incrementally as we go.

Can you prepare a pull-req with the webextension API calls, editorconfig and jshint fixes. We'll see what is left from there.

Thanks for you contribution.

**Comment by dakra** [comment]
Imho one big commit to unify all formatting is probably best
because that way you keep the history mostly in-tact.
I mean, when blaming (or looking at history), you just have
to jump that one commit.

Normally weird formatting wouldn't even bother me but
in this case it's just tabs, spaces and different
formatting mixed in the same file.


I'll make a new PR where I don't format everything but
just change the API calls, update jQuery etc and
then you can decide if you want to reformat or not :)


BTW, not sure how other ex-chrome-only extensions handle
it, but maybe even a rename to `emacs_browser` or something
could make sense ;)

File: javascript/textareas.js (hunk 1 above)
Comment (dakra) [inline javascript/textareas.js]: js prettier automatically did that change. I would guess it only changes to singlequote when there's no singlequote in the string itself so it must not escape.

**Comment by dakra** [comment]
Closing this in favour of #130

**Pull request closed without merging**